libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -263,9 +276,10 @@ void DataFlowAnalyzer::handleAssignment(std::set<YulName> const& _variables, Exp
 	}
 
 	auto const& referencedVariables = movableChecker.referencedVariables();
+	std::vector const referencedVariablesVector(referencedVariables.begin(), referencedVariables.end());
```

Comment:
```suggestion
	std::vector const referencedVariablesSorted(referencedVariables.begin(), referencedVariables.end());
```

Reply:
`m_state.references` should also have a docstring saying that the vector must be always kept sorted.

Maybe we should even rename it to `sortedReferences` to make it harder to miss that requirement? And `references()` -> `sortedReferences()`?

Reply:
Fixed

Reply:
Please also rename the vector as suggested above. I think the fact that it's sorted is more relevant here.